doc/whats_new.rst

```diff
@@ -36,7 +36,9 @@ Detailed list of changes
 🧐 API and behavior changes
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-- nothing yet
+- The package build backend has been switched from ``setuptools`` to ``hatchling``. This
+  only affects users who build and install MNE-BIDS from source, and should not lead to
+  changed runtime behavior, by `Richard Höchenberger`_ (:gh:`1204`)
```

Comment:
in what ways does it affect users who build and install mne-bids from source? :)

Reply:
I am not sure it should be in the release notes then🤔 usually we skip "infrastructure changes without effect on the user", don't we?

Reply:
There should be a Code Health section like we have in MNE-BIDS-Pipeline.

Reply:
MS does the same for the VS Code Python extension 

![image](https://github.com/mne-tools/mne-bids/assets/2046265/fbe3e543-d677-4dce-a7d0-577de1512543)